### PR TITLE
zend_hash_sort never fails since PHP 5.5.0

### PR DESCRIPTION
**Pull request by Jan-E**
See https://github.com/php/php-src/pull/3936#issuecomment-652122802

### CODE REVIEW

**Comment by Jan-E** [comment]
Compiles fine with PHP 7.4.8 x64 NTS, so should be OK.

**Comment by Jan-E** [comment]
@0mars 
The TSRMLS_CC did not do anything mote in PHP7: https://stackoverflow.com/a/49540127/872051
And they are removed in PHP8: https://github.com/php/php-src/blob/master/UPGRADING.INTERNALS#L46
They are partly in the same lines as the zend_has_sort ones, so I cannot yet make a separate PR for those changes. Shall I add them to this PR?

**Comment by 0mars** [comment]
better another PR @Jan-E

**Comment by Jan-E** [comment]
https://github.com/php/pecl-search_engine-solr/pull/21
Github says it can merge the PR

**Comment by Jan-E** [comment]
https://github.com/php/pecl-search_engine-solr/pull/21 includes both commits.
https://github.com/php/pecl-search_engine-solr/pull/20 only the zend_hash_sort one.

File: src/php7/php_solr_document.c [hunk 1]
```diff
@@ -1130,10 +1130,7 @@ PHP_METHOD(SolrDocument, sort)
 		RETURN_FALSE;
 	}
 
-	if (zend_hash_sort_ex(doc_entry->fields, zend_qsort, comparison_function, renumber TSRMLS_CC) == FAILURE) {
-
-		RETURN_FALSE;
-	}
+	zend_hash_sort_ex(doc_entry->fields, zend_qsort, comparison_function, renumber TSRMLS_CC);
```
Comment (remicollet) [inline src/php7/php_solr_document.c:1133]: Simpler to use zend_hash_sort (zend_qsort being dropped in 8)
Other way in pr #22

**Pull request closed without merging**